File: snippets/java/search_matrix_pairs.java
import static io.qdrant.client.ConditionFactory.matchKeyword;

import io.qdrant.client.QdrantClient;
import io.qdrant.client.QdrantGrpcClient;
import io.qdrant.client.grpc.Points.Filter;
import io.qdrant.client.grpc.Points.SearchMatrixPoints;

QdrantClient client =
    new QdrantClient(QdrantGrpcClient.newBuilder("localhost", 6334, false).build());

client
    .searchMatrixPairsAsync(
        SearchMatrixPoints.newBuilder()
            .setCollectionName("{collection_name}")
            .setFilter(
                Filter.newBuilder()
                    .addMust(matchKeyword("color", "red"))
                    .build()
            )
            .setSample(100)
            .setLimit(5)
            .build())
    .get();

File: snippets/csharp/search_matrix_pairs.cs
using Qdrant.Client;
using Qdrant.Client.Grpc;

var client = new QdrantClient("localhost", 6334);

await client.SearchMatrixPairsAsync(
    "{collection_name}",
    filter: MatchKeyword("color", "red"),
    sample: 100,
    limit: 5
);